### DISCUSSION TITLE
Error when trying to reinstall  Git  Backup

### DISCUSSION THREAD

**Discussion by jaxxtheone** [Guilouz/Creality-Helper-Script-Wiki#430] [Issues]
The first time when installed Git Backup I didn't find the token I had previously saved. So, I created a different repository with a new identity.  Then I removed Git Backup using Helper. 
I restarted then a new install using the new credentials. The installation, apparently, ended  successfully. 
When asked, I provided the GitHub username, my repository name and my personal access token.
That's when the new installation started acting.
I post here the error messages I got.

<img width="638" alt="Screenshot 2024-03-13 124537" src="https://github.com/Guilouz/Creality-K1-Series/assets/26436482/89ad42c5-03b8-4bca-9400-fc3c126f0c66">

**Comment by jb510** [comment]
I had problems installing git backups as well.  Tracked it down to the symlink being incorrect.   I'm away from my printer and it'd be a PITA to remotely ssh in to confirm which is which.

But if memory servers this was the symlink:
`/usr/data/printer_data/config/Helper-Script/git-backup.cfg`
and it was pointing at:
`/usr/data/helper-script/files/git-backup/git-backup.cfg/git-backup.cfg`  # note the double /git-backup.cfg/git-backup.cfg/
it should be pointing at:
`/usr/data/helper-script/files/git-backup/git-backup.cfg`

I was trying to track down the cause of this in the script but couldn't find it.

Assuming memory is correct (WARNING, I'm going from memory, test and use this at your own risk and only if you know what you're doing)
`ln -sf /usr/data/helper-script/files/git-backup/git-backup/git-backup.cfg /usr/data/printer_data/config/Helper-Script/git-backup.cfg`

**Reply by jaxxtheone** [reply]
Ouch! I'll try tonight as I'm also far from my printer. Thanks for now!

**Reply by jb510** [reply]
good luck, note I edited the above to correct my own typo

**Reply by jb510** [reply]
Looked again and found the bug. I'll try to put in a PR for it.

**Reply by jaxxtheone** [reply]
Thanks!

**Reply by jb510** [reply]
Actually looks like it's already fixed in the latest version
![CleanShot 2024-03-13 at 09 25 03@2x](https://github.com/Guilouz/Creality-K1-Series/assets/801241/bb5bb9cb-4c70-485d-8cfd-8db57c007559)


![CleanShot 2024-03-13 at 09 26 18@2x](https://github.com/Guilouz/Creality-K1-Series/assets/801241/d911748d-98b7-4d31-9759-52a6418e1834)


Weird because I had this issue last night.  Regardless fixed be @logicbus TY!

**Reply by jaxxtheone** [reply]
Concerning your last post, it didn't fix my error message. I tried a reboot between the 'remove' and 'install' again (16) Git Backup but something is left after an unsuccessful install. What I do think is that removing leaves some footprints around. Tonight I'll try the symlink fix you suggested above.

**Reply by jb510** [reply]
The other thing I did was go to 
/usr/data/printer_data/config/ (again going from memory that that is where the actually git repo (.git folder) is)

And make sure git config was correct, then manually push it the first time because it hadn't actually created a main branch.

`git push --set-upstream origin master`

My guess is maybe the previous error broke install.  After fixing the link, I'd re-run the git backup install.  If that STILL doesn't work, maybe try the above.

**Reply by BroMarduk** [reply]
I also had to edit my .env config file from within Fluidd and set the BRANCH=master for things to work without seeing errors from the macros.

I wish we could set the BRANCH in the installer script like we do the user name, repository and token.

**Reply by jaxxtheone** [reply]
OK, gentlemen!
My .env  reads:
```
IFS=/usr/data/printer_data/config
GITHUB_TOKEN=******token*****
REMOTE=K1_RCK
BRANCH=
USER=jaxxtheone
```
Yesterday I was busy, today I'll try to fix this up.
Thanks!

**Reply by jaxxtheone** [reply]
Still no joy. It should be my mistake or lack of knowledge. If someone can recommend me a tutorial I'm eager to learn. It seems easy but I'm misunderstanding something about the process. of automatic upload. I can do it manually (what I'm doing now) but I can't do it the way Guillouz designed it. My .git at /usr/data/printer_data/config/ looks odd.
Any help will be welcome.

```
[core]
	repositoryformatversion = 0
	filemode = true
	bare = false
	logallrefupdates = true
[remote "origin"]
	url = https://jaxxtheone:***token***@github.com/jaxxtheone/K1_RCK.git
	fetch = +refs/heads/*:refs/remotes/origin/*
```

**Comment by jb510** [comment]
@jaxxtheone 

You haven't set the remote branch
Please try this.
```
cd /usr/data/printer_data/config/
git push --set-upstream origin master
``` 
 
![CleanShot 2024-03-15 at 09 02 42@2x](https://github.com/Guilouz/Creality-K1-Series/assets/801241/386d6a95-a896-462b-bf8c-46fd6b6cb7d7)

I'm not sure off hand what .env is used for in this case, I'm guessing the script stores things there for the script to access, but /.git/config is what should matter to git actually functioning.

**Reply by jaxxtheone** [reply]
Doing it tonight!, thanks!
Indeed it makes total sense, I bet my "config" is faulty, I hope this will do the magic.

**Reply by Horgul123** [reply]
thanks for your hint, this helped a little and your command run withou error. But if i try to backup from the script I'm getting

`Enumerating objects: 5, done.
Counting objects: 100% (5/5), done.
Delta compression using up to 2 threads
Compressing objects: 100% (2/2), done.
Writing objects: 100% (3/3), 289 bytes | 96.00 KiB/s, done.
Total 3 (delta 1), reused 0 (delta 0)
remote: Resolving deltas: 100% (1/1), completed with 1 local object.
To https://github.com/Horgul123/k1_backup.git
   6edeb64..1f01f36  master -> master
/usr/data/helper-script/files/git-backup/git-backup.sh: line 304: pkill: not found
[master bdb6744] Auto-commit on 26-03-2024 (20:31:22) by Git Backup
 Committer: root <root@K1-D35C.fritz.box>
Your name and email address were configured automatically based
on your username and hostname. Please check that they are accurate.
You can suppress this message by setting them explicitly:

    git config --global user.name "Your Name"
    git config --global user.email you@example.com

After doing this, you may fix the identity used for this commit with:

    git commit --amend --reset-author

 1 file changed, 0 insertions(+), 0 deletions(-)
Push command is git push origin
Enumerating objects: 5, done.
Counting objects: 100% (5/5), done.
Delta compression using up to 2 threads
Compressing objects: 100% (3/3), done.
Writing objects: 100% (3/3), 201.19 KiB | 2.19 MiB/s, done.
Total 3 (delta 1), reused 0 (delta 0)
remote: Resolving deltas: 100% (1/1), completed with 1 local object.
To https://github.com/Horgul123/k1_backup.git
   1f01f36..bdb6744  master -> master
/usr/data/helper-script/files/git-backup/git-backup.sh: line 304: pkill: not found
`


any ideas?
